L1Trigger/VertexFinder/plugins/VertexProducer.cc

```diff
@@ -57,21 +59,41 @@ VertexProducer::VertexProducer(const edm::ParameterSet& iConfig)
     case Algorithm::Kmeans:
       edm::LogInfo("VertexProducer") << "VertexProducer::Finding vertices using a kmeans algorithm";
       break;
+    case Algorithm::Generator:
+      edm::LogInfo("VertexProducer")
+          << "VertexProducer::Finding vertices using ** GENERATOR ** vertex (average of TP z0s)";
+      break;
+    case Algorithm::NNEmulation:
+      edm::LogInfo("VertexProducer") << "VertexProducer::Finding vertices using the Neural Network Emulation";
+      break;
   }
 
   //--- Define EDM output to be written to file (if required)
-  if (settings_.vx_algo() == Algorithm::fastHistoEmulation) {
+  if ((settings_.vx_algo() == Algorithm::fastHistoEmulation) | (settings_.vx_algo() == Algorithm::NNEmulation)) {
```

Comment:
Should this really be bitwise OR? I would have thought `||`

Reply:
Done, changed to ||